pontoon/base/models.py

```diff
@@ -1148,6 +1158,9 @@ def for_project_locale(self, project, locale, paths=None, filter_type=None,
             elif filter_type == 'unchanged':
                 entities = self.objects.unchanged(locale)
 
+            elif filter_type in Translation.authors(locale, project, paths).values_list('email', flat=True):
```

Comment:
Yes, that's a possible optimization. It's also slightly less powerful than this check. For now I'll leave it, but it's good to know we can be faster here.